crates/chia-puzzles/src/puzzles/nft.rs

```diff
@@ -24,6 +46,32 @@ pub struct NftStateLayerArgs<I, M> {
     pub inner_puzzle: I,
 }
 
+impl<I, M> NftStateLayerArgs<I, M> {
+    pub fn new(metadata: M, inner_puzzle: I) -> Self {
+        Self {
+            mod_hash: NFT_STATE_LAYER_PUZZLE_HASH.into(),
```

Comment:
I suppose we'd want to make `mod_hash` a `TreeHash` at some point, right?
(as well as `metadata_updater_puzzle_hash` and `launcher_puzzle_hash`)

Reply:
No, since these need to be encoded as atoms rather than the value of the tree hash's preimage.